### PR TITLE
Provide conditional usage for 'setupMiddlewareReporter'

### PR DESCRIPTION
**Pull request by drewlee**
Base implementation to address https://github.com/ember-a11y/ember-a11y-testing/issues/266.

The basic idea is to output the environment variables in `addon-test-support/cli-options.ts`. If either one is specified, their corresponding value is overridden at build-time to `true`, and thus accessible from the browser/headless-browser environment.

This also provides a new `useMiddlewareReporter` API, to conditionally invoke `setupMiddlewareReporter`.

```javascript
import { useMiddlewareReporter, setupMiddlewareReporter } from 'ember-a11y-testing/test-support';

if (useMiddlewareReporter()) {
  setupMiddlewareReporter();
}
```

Either one of these invocations will make `useMiddlewareReporter` return `true`, otherwise it's `false`:
```bash
ember test --query "enableA11yMiddlewareReporter=true"
ENABLE_A11Y_MIDDLEWARE_REPORTER=true ember test
```

Either one of these invocations will force a11y audit:
```bash
ember test --query "enableA11yAudit=true"
ENABLE_A11Y_AUDIT=true ember test
```

Ran this implementation with several apps to verify builds and tests work as expected.

### CODE REVIEW

**Comment by scalvert** [comment]
Is this still a draft or do you think it's ready, @drewlee?

File: addon-test-support/index.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ export { default as a11yAudit } from './audit';
 export { default as a11yAuditIf } from './audit-if';
 export { setRunOptions, getRunOptions } from './run-options';
 export { setEnableA11yAudit, shouldForceAudit } from './should-force-audit';
+export { shouldUseMiddlewareReporter } from './should-use-middleware-reporter';
```
Comment (scalvert) [inline addon-test-support/index.ts:5]: What do you think about renaming this to `useMiddlewareReporter` to keep it more terse?

File: addon-test-support/should-use-middleware-reporter.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { ENABLE_A11Y_MIDDLEWARE_REPORTER } from './cli-options';
```
Comment (scalvert) [inline addon-test-support/should-use-middleware-reporter.ts:1]: If we rename this to `useMiddlewareReporter`, should we also rename this const to `USE_A11Y_MIDDLEWARE_REPORTER`?

File: addon-test-support/index.ts (hunk 1 above)
Comment (drewlee) [reply addon-test-support/index.ts:5]: Yeah, that reads much better. I'll update.

File: addon-test-support/should-use-middleware-reporter.ts (hunk 2 above)
Comment (drewlee) [reply addon-test-support/should-use-middleware-reporter.ts:1]: I'd rather keep it as is so it's consistent with `ENABLE_A11Y_AUDIT` and `enableA11yAudit=true`.

File: README.md [hunk 3]
```diff
@@ -260,7 +260,7 @@ test('Some test case', async function (assert) {
 ### Force Running audits
 
 `ember-a11y-testing` allows you to force audits if `enableA11yAudit` is set as a query param
-on the test page. This is useful if you want to conditionally run accessibility audits, such
+on the test page or the `ENABLE_A11Y_AUDIT` environmental variable is provided. This is useful if you want to conditionally run accessibility audits, such
```
Comment (scalvert) [inline README.md:263]:
```suggestion
on the test page or the `ENABLE_A11Y_AUDIT` environment variable is provided. This is useful if you want to conditionally run accessibility audits, such
```

File: README.md [hunk 4]
```diff
@@ -349,6 +349,28 @@ setupMiddlewareReporter();
 start();
 ```
 
+A helper function is available to use the middleware reporter conditionally, allowing interoperability between the default reporter and the middleware reporter. Import `useMiddlewareReporter` and apply as a check around the `setupMiddlewareReporter` function in `tests/test-helper.js`. The middleware reporter will now only be invoked when `enableA11yMiddlewareReporter` is set as a query param on the test page or the `ENABLE_A11Y_MIDDLEWARE_REPORTER` environmental variable is provided.
```
Comment (scalvert) [inline README.md:352]:
```suggestion
A helper function is available to use the middleware reporter conditionally, allowing interoperability between the default reporter and the middleware reporter. Import `useMiddlewareReporter` and apply as a check around the `setupMiddlewareReporter` function in `tests/test-helper.js`. The middleware reporter will now only be invoked when `enableA11yMiddlewareReporter` is set as a query param on the test page or the `ENABLE_A11Y_MIDDLEWARE_REPORTER` environment variable is provided.
```

File: README.md [hunk 5]
```diff
@@ -349,6 +349,28 @@ setupMiddlewareReporter();
...
+if (useMiddlewareReporter()) {
+  // Only runs if `enableA11yMiddlewareReporter` is set in URL
+  setupMiddlewareReporter();
+}
+
+start();
+```
+
+Note, as a convenience, `useMiddlewareReporter` automatically forces audits, thus explicitly specifying
+the `enableA11yAudit` query param or the `ENABLE_A11Y_AUDIT` environmental variable is unnecessary.
```
Comment (scalvert) [inline README.md:372]:
```suggestion
the `enableA11yAudit` query param or the `ENABLE_A11Y_AUDIT` environment variable is unnecessary.
```

File: addon-test-support/should-force-audit.ts [hunk 6]
```diff
@@ -14,13 +16,14 @@ export function setEnableA11yAudit(enabled: boolean = false) {
 
 /**
  * Forces running audits. This functionality is enabled by
- * the presence of an `enableA11yAudit` query parameter passed to the test suite.
+ * the presence of an `enableA11yAudit` query parameter passed to the test suite
+ * or the `ENABLE_A11Y_AUDIT` environmental variable.
```
Comment (scalvert) [inline addon-test-support/should-force-audit.ts:20]:
```suggestion
 * or the `ENABLE_A11Y_AUDIT` environment variable.
```

**Review by scalvert — CHANGES_REQUESTED** [review]

**Comment by scalvert** [comment]
Overall this looks good. Thanks, @drewlee!

**Review by scalvert — APPROVED** [review]

**Pull request merged**